PSReadLine/ReadLine.cs

```diff
@@ -292,10 +279,12 @@ internal static PSKeyInfo ReadKey()
                 throw new OperationCanceledException();
             }
 
-            if (handleId == CancellationRequested)
+            if (_singleton._cancelReadCancellationToken.IsCancellationRequested)
```

Comment:
There is a race condition that we reach here due to a timeout from `WaitHandle.WaitAny`, but then this check returns `true`. In this case, `Deque()` may raise an `Queue empty` exception.